### PR TITLE
Marketplace search

### PR DESCRIPTION
**Pull request by abocquet**
Add search in marketplace

### CODE REVIEW

File: frontend/src/components/utils/Pagination.tsx [hunk 1]
```diff
@@ -61,6 +61,21 @@ export const Pagination = ({
...
     );
 
+    // When we change the query params there my be a different number of
+    // results. Thus we need when the key change, to reset the page to 1.
+    // However we shouldn't do it before the key is initialize, hence the
+    // prevApiKey !== ""
+    const [prevApiKey, setPrevApiKey] = useState("");
+    useEffect(() => {
+        let apiKeyJSON = JSON.stringify(apiKey);
+        if (apiKeyJSON !== prevApiKey) {
```
Comment (Piwakk) [inline frontend/src/components/utils/Pagination.tsx:71]: I know this is a very sensitive topic but you write `Api` but `JSON`. Please choose a standard.
I am not sure if there are similar names in the codebase which could give us a guideline.
FWIW, my inclination goes to `prevApiKey` and `apiKeyJson`.

File: frontend/src/components/utils/Pagination.tsx [hunk 2]
```diff
@@ -61,6 +61,21 @@ export const Pagination = ({
         config
     );
 
+    // When we change the query params there my be a different number of
+    // results. Thus we need when the key change, to reset the page to 1.
+    // However we shouldn't do it before the key is initialize, hence the
+    // prevApiKey !== ""
+    const [prevApiKey, setPrevApiKey] = useState("");
```
[the remark this answers is not in the document]
Comment (Piwakk) [reply frontend/src/components/utils/Pagination.tsx:68]: Not a big deal, but you should mention that `prevApiKey` contains stringified JSON.

[the remark this answers is not in the document]
Comment (abocquet) [reply frontend/src/components/utils/Pagination.tsx:68]: Removed in #205 anyway

File: frontend/src/components/utils/Pagination.tsx (hunk 1 above)
Comment (abocquet) [reply frontend/src/components/utils/Pagination.tsx:71]: Same

File: frontend/src/components/utils/sidebar/SidebarInputSearch.tsx [hunk 3]
```diff
@@ -1,6 +1,19 @@
 import React from "react";
 import { DebounceInput } from "react-debounce-input";
 
+/**
+ * Debounced input field for the sidebar
+ * @param setParams the useState function to update the `search` param
```
Comment (Piwakk) [inline frontend/src/components/utils/sidebar/SidebarInputSearch.tsx:6]: It does not _have_ to be a function from `useState`, maybe use a more generic phrasing, like a "callback"

File: frontend/src/components/utils/sidebar/SidebarUserSearch.tsx [hunk 4]
```diff
@@ -12,7 +12,21 @@ const promiseOptions = (inputValue) =>
         }))
     );
 
-export const SidebarUserSearch = ({ setParams, userName = "", ...props }) => (
+/**
+ * Debounced input field for the sidebar
+ * @param setParams the useState function to update the `search` param
+ * @param paramKey the name of the parameter in the API query. "user" by default
```
Comment (Piwakk) [inline frontend/src/components/utils/sidebar/SidebarUserSearch.tsx:18]: I think `apiKey` would be more explicit

File: frontend/src/components/utils/sidebar/SidebarUserSearch.tsx [hunk 5]
```diff
@@ -12,7 +12,21 @@ const promiseOptions = (inputValue) =>
         }))
     );
 
-export const SidebarUserSearch = ({ setParams, userName = "", ...props }) => (
+/**
+ * Debounced input field for the sidebar
```
Comment (Piwakk) [inline frontend/src/components/utils/sidebar/SidebarUserSearch.tsx:16]: This is a debounced field for searching users, please be more specific.

File: frontend/src/components/utils/sidebar/SidebarUserSearch.tsx [hunk 6]
```diff
@@ -12,7 +12,21 @@ const promiseOptions = (inputValue) =>
...
+ * @constructor
+ *
+ * Usage example :
+ * [search, setSearch] = useState({});
+ * return <SidebarInputSearch setParams={setSearch} paramKey="buyer"/>
+ *
+ * The value of search after "something" was typed is :
+ * {buyer: something}
+ */
+export const SidebarUserSearch = ({ setParams, paramKey = "", ...props }) => (
```
Comment (Piwakk) [inline frontend/src/components/utils/sidebar/SidebarUserSearch.tsx:29]:
```suggestion
export const SidebarUserSearch = ({ setParams, paramKey = "user", ...props }) => (
```

**Pull request merged**